src/modalities/models/model_factory.py

```diff
@@ -631,7 +640,7 @@ def get_gpt2_tensor_parallelized_model(model: GPT2LLM, device_mesh: DeviceMesh)
             ),
         }
 
-        if isinstance(model.transformer.wpe, nn.Embedding):
+        if hasattr(model.transformer, "wpe") and isinstance(model.transformer.wpe, nn.Embedding):
```

Comment:
I checked the gpt2_model implementation. `model.transformer.wpe` should be always present either as an embedding or identity layer. Do we really need this check?